fix: Allow displayed folder share modal to close when going backwards

cballevre opened the pull request:
The main idea behind this PR is to allow the sharing modal to close when you go back, which is useful in the Flagship application. This only applies the share button for the displayed folder

```
### 🐛 Bug Fixes

* Allow displayed folder share modal to close when going backwards
* Show the file when accessing to `/file/:fileId`

### 🔧 Tech

* Update react-router-dom from 6.6.0 to 6.14.2
```

Crash-- commented on src/drive/web/modules/views/Modal/ShareView.jsx:
```diff
@@ -0,0 +1,31 @@
+  if (displayedFolder) {
+    const onClose = () => {
+      // navigate(`/folder/${displayedFolder.id}`)
+      navigate(-1)
```
Don't use -1, we have issue with this on the flagship app. You can use `..` or the path you want, but not -1 🙏

cballevre replied:
It's replaced

Crash-- commented:
Open question: Why this route? Can you add a bit of context in this PR? Also, why a `/share/` route and not `/file/:id/share`? With the latest, each file has its own route, and we can copy past it, like that, we don't need to rely on the selectedFolder.

cballevre commented on src/drive/web/modules/drive/Toolbar/share/ShareButton.jsx:
```diff
@@ -1,14 +1,20 @@
+  const { pathname } = useLocation()
+
+  const share = () => {
+    navigate(`${pathname}/share`)
```
It should work, but I don't know why it doesn't work on cozy-drive. I'm trying to work out where the problem might be

cballevre replied:
I didn't found why it doesn't work in this environnement

cballevre commented:
I've added a bit of context to the PR description and renamed the commit name to be more explicit.

I didn't use `file/:fileId/share` because it's a route to share a file. For the moment, it is not used. The modal is called directly in the table but I imagine it will be soon. I also didn't want to mix up the concept of folder and file for the user in the access path
